source/common/buffer/buffer_impl.cc

```diff
@@ -47,6 +47,48 @@ void OwnedImpl::addBufferFragment(BufferFragment& fragment) {
   slices_.emplace_back(fragment);
 }
 
+std::pair<bool, size_t> findDequeIndex(size_t i, RawSlice& slice, SliceDeque& slices) {
+  size_t empty_slices = 0;
+  size_t deque_index = 0;
+
+  // This is the same logic used by getRawSlices require any changes upstream applied here too
+  for (const auto& deque_slice : slices) {
+    if (deque_slice.dataSize() == 0) {
+      ++empty_slices;
+      continue;
+    }
+    if (deque_index == i && deque_slice.data() == slice.mem_ &&
```

Comment:
What would be the case where `deque_index == i`  it doesn't match the slice given that `getRawSlices()` also uses this logic? In that case why keep iterating and not just return?

Can we get away with using just the RawSlice?